ngraph/core/reference/include/ngraph/runtime/reference/convolution_backprop_data.hpp

```diff
@@ -0,0 +1,309 @@
+                    //   (N,0,s_1*i_1,s_2*i_2,...,s_n*i_n) ->
+                    //
+                    //     (N+1,
+                    //      chans_in_count,
+                    //      s_1*i_1+ l_1*filter_dims_1,
+                    ///       ...,
+                    ///     s_n*i_n +l_n*filter_dims_n)
+                    //
+                    // with strides:
+                    //
+                    //   (1,l_1,...,l_n).
+                    //
+                    // Note that we are iterating within the *padded* and *dilated* in batch, so
+                    // further down we must check the current coordinate is in the pad or dilation
+                    // gap.
+
+                    size_t n_spatial_dimensions = in_shape.size() - 2;
+                    size_t n_in_channels = in_shape[in_channel_axis];
+
+                    Coordinate in_transform_start(2 + n_spatial_dimensions);
+                    Coordinate in_transform_end(2 + n_spatial_dimensions);
+                    Strides in_transform_movement_strides(2 + n_spatial_dimensions, 1);
+                    CoordinateDiff in_transform_pad_below(2 + n_spatial_dimensions, 0);
+                    CoordinateDiff in_transform_pad_above(2 + n_spatial_dimensions, 0);
+                    Strides in_transform_dilation_strides(2 + n_spatial_dimensions, 1);
+
+                    in_transform_start[in_batch_axis] = batch_index;
+                    in_transform_end[in_batch_axis] = batch_index + 1;
+                    in_transform_start[in_channel_axis] = 0;
+                    in_transform_end[in_channel_axis] = 1;
+
+                    for (size_t i = 2; i < n_spatial_dimensions + 2; i++)
+                    {
+                        size_t filter_dilation_stride = filter_dilation[i - 2];
+                        size_t filter_movement_stride = stride[i - 2];
+                        std::ptrdiff_t below_pad = in_pad_below[i - 2];
+                        std::ptrdiff_t above_pad = in_pad_above[i - 2];
+                        size_t in_dilation_stride = in_dilation[i - 2];
+
+                        in_transform_start[i] = filter_movement_stride * out_coord[i];
+                        in_transform_end[i] = in_transform_start[i] +
+                                              (filter_shape[i] - 1) * filter_dilation_stride + 1;
+                        in_transform_movement_strides[i] = filter_dilation_stride;
+                        in_transform_pad_below[i] = below_pad;
+                        in_transform_pad_above[i] = above_pad;
+                        in_transform_dilation_strides[i] = in_dilation_stride;
+                    }
+
+                    AxisVector in_transform_axis_order(2 + n_spatial_dimensions);
+                    for (size_t i = 0; i < in_transform_axis_order.size(); i++)
+                    {
+                        in_transform_axis_order[i] = i;
+                    }
+                    CoordinateTransform in_transform(in_shape,
```

Comment:
ConvolutionBackpropData will be addressed in different ticket, so do not review it's code - I didn't change it, only moved it to other file because previously it was together with convolution.